core/src/main/java/nl/nn/adapterframework/http/rest/ApiListener.java

```diff
@@ -221,39 +229,20 @@ public boolean getUpdateEtag() {
 	//TODO add authenticationType
 
 	@IbisDoc({"7", "enables security for this listener, must be one of [NONE, COOKIE, HEADER, AUTHROLE]. If you wish to use the application servers authorisation roles [AUTHROLE], you need to enable them globally for all ApiListeners with the `servlet.ApiListenerServlet.securityroles=ibistester,ibiswebservice` property", "NONE"})
-	public void setAuthenticationMethod(String authenticationMethod) throws ConfigurationException {
-		try {
-			this.authenticationMethod = AuthenticationMethods.valueOf(authenticationMethod);
-		}
-		catch (IllegalArgumentException iae) {
-			throw new ConfigurationException("Unknown authenticationMethod ["+authenticationMethod+"]. Must be one of "+ Arrays.asList(AuthenticationMethods.values()));
-		}
+	public void setAuthenticationMethod(String authenticationMethod) {
+		this.authenticationMethod = authenticationMethod;
 	}
 
-	public AuthenticationMethods getAuthenticationMethod() {
+	public AuthenticationMethods getAuthenticationMethodType() {
 		if(authenticationMethod == null) {
-			authenticationMethod = AuthenticationMethods.NONE;
+			return AuthenticationMethods.NONE;
 		}
```

Comment:
All you had to do was rename the method, the logic could have stayed where it was.

The idea of enums is that once (in the near future?) we can add them to our xsd so people can use autocomplete to fill the attribute values.

There isn't really a standard yet, but I think we should use `getMethodNameEnum()` so in this case `getAuthenticationMethodEnum`. Since there is no need to have a getter, and we don't want the 'BeansParser' to error when the dataTypes are different, just renaming the method should be enough. 

If a getter is desirable, the getter-method should call the `getMethodNameEnum().name()` method, which in turn returns the correct dataType.